<?xml version="1.0"?>
<launch>
    <!-- <args for urdf/> -->
    <arg name="camera_model"      default="OAK-D" /> <!-- 'zed' or 'zedm' -->
    <arg name="tf_prefix"         default="oak" />
    <arg name="mode"              default="depth" />
    <arg name="base_frame"        default="oak-d_frame" />
    <arg name="parent_frame"      default="oak-d-base-frame" />

    <arg name="cam_pos_x"         default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"         default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"         default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"          default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"         default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"           default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    
    <arg name="confidence"        default="240" /> 
    <arg name="LRchecktresh"      default="5" /> 
    <arg name="lrcheck"           default="true" /> 
    <arg name="extended"          default="false" /> 
    <arg name="subpixel"          default="true" />
    <arg name="monoResolution"    default="400p" /> 
    <arg name="camera_param_uri"  default="package://rpi_node/params/camera" />

   
    <include file="$(find depthai_bridge)/launch/urdf.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="parent_frame"    value="$(arg  parent_frame)"/>
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="tf_prefix"       value="$(arg  tf_prefix)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)"   />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)"   />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)"   />
        <arg name="cam_roll"        value="$(arg  cam_roll)"    />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)"   />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"     />
    </include>

    <!-- launch-prefix="xterm -e gdb (add [- - args] without space) -->
    <node name="imu_rgb_stereo" pkg="rpi_node" type="imu_rgb_stereo_publisher2" output="screen" required="true">
        <param name="tf_prefix"        value="$(arg tf_prefix)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>
        <param name="mode"             value="$(arg mode)"/>
        <param name="lrcheck"          value="$(arg lrcheck)"/>
        <param name="extended"         value="$(arg extended)"/>
        <param name="subpixel"         value="$(arg subpixel)"/>
        <param name="confidence"       value="$(arg confidence)"/>
        <param name="LRchecktresh"     value="$(arg LRchecktresh)"/>
        <param name="monoResolution"   value="$(arg monoResolution)"/>
    </node>            

   
</launch>

